oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,313 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+
+#include "../sandboxed.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/sandbox2/util/path.h"
+#include "sandboxed_api/vars.h"
+#include "tiffio.h"  // NOLINT(build/include)
+
+namespace {
+
+struct ChannelLimits {
+  uint8_t min_red;
+  uint8_t max_red;
+  uint8_t min_green;
+  uint8_t max_green;
+  uint8_t min_blue;
+  uint8_t max_blue;
+  uint8_t min_alpha;
+  uint8_t max_alpha;
+};
+
+constexpr unsigned kRawTileNumber = 9;
+constexpr unsigned kClusterSize = 6;
+constexpr unsigned kChannelsInPixel = 3;
+constexpr unsigned kTestCount = 3;
+constexpr unsigned kImageSize = 128 * 128;
+constexpr unsigned kClusterImageSize = 64 * 64;
+using ClusterData = std::array<uint8_t, kClusterSize>;
+
+constexpr std::array<std::pair<unsigned, ClusterData>, kTestCount> kClusters = {
+    {{0, {0, 0, 2, 0, 138, 139}},
+     {64, {0, 0, 9, 6, 134, 119}},
+     {128, {44, 40, 63, 59, 230, 95}}}};
+
+constexpr std::array<std::pair<unsigned, ChannelLimits>, kTestCount> kLimits = {
+    {{0, {15, 18, 0, 0, 18, 41, 255, 255}},
+     {64, {0, 0, 0, 0, 0, 2, 255, 255}},
+     {512, {5, 6, 34, 36, 182, 196, 255, 255}}}};
+
+absl::Status CheckCluster(unsigned cluster,
+                          const sapi::v::Array<uint8_t>& buffer,
+                          const ClusterData& expected_cluster) {
+  if (buffer.GetSize() <= cluster * kClusterSize) {
+    return absl::InternalError("Buffer overrun\n");
+  }
+  auto target = buffer.GetData() + cluster * kClusterSize;
+
+  if (!std::memcmp(target, expected_cluster.data(), kClusterSize)) {
+    return absl::OkStatus();
+  }
+
+  // the image is split on 6-bit clusters because it has YCbCr color format
+  return absl::InternalError(absl::StrCat(
+      "Cluster ", cluster, " did not match expected results.\n", "Expect: ",
+      expected_cluster[0], "\t", expected_cluster[1], "\t", expected_cluster[2],
+      "\t", expected_cluster[3], "\t", expected_cluster[4], "\t",
+      expected_cluster[5], "\n", "Got: ", target[0], "\t", target[1], "\t",
+      target[2], "\t", target[3], "\t", target[4], "\t", target[5], "\n"));
+}
+
+absl::Status CheckRgbPixel(unsigned pixel, const ChannelLimits& limits,
+                           const sapi::v::Array<uint8_t>& buffer) {
+  if (buffer.GetSize() <= pixel * kChannelsInPixel) {
+    return absl::InternalError("Buffer overrun\n");
+  }
+  auto rgb = buffer.GetData() + kChannelsInPixel * pixel;
```

Comment:
use `auto *rgb` (explicitly marking this as a pointer)

Reply:
That should be `auto* rgb`, keeping the asterisk closer to the type :)

The project `.clang-format` file automatically fixes this.

Reply:
It is still not a `auto* rgb` (mind the star)